client/src/pages/admin/mentor-registry.tsx

```diff
@@ -1,40 +1,54 @@
-import { getCoursesProps as getServerSideProps } from 'modules/Course/data/getCourseProps';
-import { GithubUserLink } from 'components/GithubUserLink';
-import { colorTagRenderer, getColumnSearchProps, stringSorter, tagsRenderer } from 'components/Table';
-import { useLoading } from 'components/useLoading';
-import CopyToClipboardButton from 'components/CopyToClipboardButton';
-import { Session, withSession } from 'components/withSession';
-import { useCallback, useState } from 'react';
+import { useCallback, useState, useMemo, useEffect } from 'react';
 import { useAsync } from 'react-use';
+
+import { FileExcelOutlined } from '@ant-design/icons';
+import { Alert, Button, Col, message, notification, Row, Tabs, Typography } from 'antd';
+
+import { DisciplineDto, DisciplinesApi, MentorRegistryDto } from 'api';
+
 import { CoursesService } from 'services/courses';
 import { MentorRegistryService } from 'services/mentorRegistry';
 import { Course } from 'services/models';
-import css from 'styled-jsx/css';
+
+import { MentorRegistryTable } from 'modules/MentorRegistry/components/MentorRegistryTable';
+import { MentorRegistryModal } from 'modules/MentorRegistry/components/MentorRegistryModal';
+import { MentorRegistryTableContainer } from 'modules/MentorRegistry/components/MentorRegistryTableContainer';
+import { MentorRegistryTabsMode } from 'modules/MentorRegistry/constants';
+import { getCoursesProps as getServerSideProps } from 'modules/Course/data/getCourseProps';
+
+import { useLoading } from 'components/useLoading';
+import { Session, withSession } from 'components/withSession';
 import { AdminPageLayout } from 'components/PageLayout';
-import { MentorRegistryDto } from 'api';
+import { tabRenderer } from 'components/TabsWithCounter/renderers';
 
-const { Content } = Layout;
-const PAGINATION = 200;
+import css from 'styled-jsx/css';
+
+type Props = {
+  courses: Course[];
+  session: Session;
+};
+
+type NotificationType = 'success' | 'info' | 'warning' | 'error';
 
-type Props = { courses: Course[]; session: Session };
 const mentorRegistryService = new MentorRegistryService();
 const coursesService = new CoursesService();
+const disciplinesApi = new DisciplinesApi();
 
 function Page(props: Props) {
   const [loading, withLoading] = useLoading(false);
 
+  const [api, contextHolder] = notification.useNotification();
+
   const [modalLoading, setModalLoading] = useState(false);
   const [showAll, setShowAll] = useState(false);
-
   const [data, setData] = useState<MentorRegistryDto[]>([]);
   const [allData, setAllData] = useState<MentorRegistryDto[]>([]);
   const [maxStudents, setMaxStudents] = useState(0);
-
   const [courses, setCourses] = useState<Course[]>([]);
   const [modalData, setModalData] = useState(null as Partial<any> | null);
+  const [activeTab, setActiveTab] = useState<MentorRegistryTabsMode>(MentorRegistryTabsMode.New);
+  const [disciplines, setDisciplines] = useState([] as DisciplineDto[]);
```

Comment:
It would be better to get rid of `any` here, but anyway at least we should avoid typecast  
```suggestion
  const [modalData, setModalData] = useState<any>(null);
  const [activeTab, setActiveTab] = useState<MentorRegistryTabsMode>(MentorRegistryTabsMode.New);
  const [disciplines, setDisciplines] = useState<DisciplineDto[]>([]);
```